contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,491 @@
+        staking.unwrap(msg.sender, amountWithdrawn);
+    }
+
+    /**
+        @notice Withdraw all gOHM from all donor positions
+    */
+    function withdrawAll() external {
+        if (withdrawDisabled) revert YieldDirector_WithdrawalsDisabled();
+
+        uint256[] storage depositIds = depositorIds[msg.sender];
+
+        uint256 depositsLength = depositIds.length;
+        if (depositsLength == 0) revert YieldDirector_NoDeposits();
+
+        uint256 principalTotal = 0;
+
+        for (uint256 index = 0; index < depositsLength; ++index) {
+            DepositInfo storage currDeposit = depositInfo[depositIds[index]];
+
+            principalTotal += currDeposit.principalAmount;
+
+            _withdrawAllPrincipal(currDeposit.id);
+        }
+
+        IERC20(gOHM).safeTransfer(msg.sender, _toAgnostic(principalTotal));
+
+        emit AllWithdrawn(msg.sender, _toAgnostic(principalTotal));
+    }
+
+    /**
+        @notice Get deposited gOHM amounts for specific recipient (updated to current index
+                based on sOHM equivalent amount deposit)
+        @param donor_ Address of user donating yield
+        @param recipient_ Address of user receiving donated yield
+    */
+    function depositsTo(address donor_, address recipient_) external view returns (uint256) {
+        uint256[] storage depositIds = depositorIds[donor_];
+
+        for (uint256 index = 0; index < depositIds.length; ++index) {
+            uint256 id = depositIds[index];
+
+            if (recipientLookup[id] == recipient_) {
+                return _toAgnostic(depositInfo[id].principalAmount);
+            }
+        }
+
+        return 0;
+    }
+
+    /**
+        @notice Return total amount of donor's gOHM deposited (updated to current index based
+                on sOHM equivalent amount deposited)
+        @param donor_ Address of user donating yield
+    */
+    function totalDeposits(address donor_) external view returns (uint256) {
+        uint256[] storage depositIds = depositorIds[donor_];
+        uint256 principalTotal = 0;
+
+        for (uint256 index = 0; index < depositIds.length; ++index) {
+            principalTotal += depositInfo[depositIds[index]].principalAmount;
+        }
+
+        return _toAgnostic(principalTotal);
+    }
+
+    /**
+        @notice Return arrays of donor's recipients and deposit amounts (gOHM value based on
+                sOHM equivalent deposit), matched by index
+        @param donor_ Address of user donating yield
+    */
+    function getAllDeposits(address donor_) external view returns (address[] memory, uint256[] memory) {
+        uint256[] storage depositIds = depositorIds[donor_];
+
+        uint256 len = depositIds.length == 0 ? 1 : depositIds.length;
+
+        address[] memory addresses = new address[](len);
+        uint256[] memory agnosticDeposits = new uint256[](len);
+
+        if (depositIds.length == 0) {
+            addresses[0] = address(0);
```

Comment:
do we need this? i though address(o) and 0 are default values for address and unsigned integers

Reply:
i guess its a external view functions so gas isn't really that important here

Reply:
think you can leave it, might be more clear what it does if you leave it.